ontopy/ontology.py

```diff
@@ -922,36 +931,44 @@ def save(
                     "'Known issues' section of the README."
                 )
             )
-
         revmap = {value: key for key, value in FMAP.items()}
         if filename is None:
             if format:
                 fmt = revmap.get(format, format)
-                filename = f"{self.name}.{fmt}"
+                file = f"{self.name}.{fmt}"
             else:
                 raise TypeError("`filename` and `format` cannot both be None.")
-        filename = os.path.join(dir, filename)
-        dir = Path(filename).resolve().parent
+        else:
+            file = filename
+        filepath = os.path.join(dir, file)
+        returnpath = filepath
```

Comment:
One could consider making `filepath` a pathlib.Path object. That would also mean that save() will return a Path object instead of a string, which would be little more explicit and potentially little more cross-platform.

Reply:
Done